src/services/download-service.ts

```diff
@@ -70,20 +73,47 @@ class DownloadService {
       const downloadUrl = response.url;
 
       const isContentTypeSupported = this.player.isImage() || this.isContentTypeSupported(response);
-      const fileName = this.player.isImage() ? this.player.sources?.metadata?.name || 'Image' : this.getFilename(response);
+      const fileName = this.getFilename(response);
 
       if (isContentTypeSupported && fileName) {
         return {
           downloadUrl,
           fileName
         };
       }
-    } catch (e: any) {}
+    } catch (e: any) {
+      // in case HEAD request failed for raw service (image use-case), retry to get the image download metadata using thumbnail service
+      if (this.player.isImage()) {
+        return await this.getImageDownloadMetadata();
+      }
+    }
+
+    return null;
+  }
 
+  private async getImageDownloadMetadata(): Promise<DownloadMetadata> {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const imageSource = this.player.config.sources.image;
+    let requestUrl = imageSource && imageSource.length > 0 ? imageSource[0].url : '';
+    if (!requestUrl) return null;
+    if (!requestUrl.includes('/ks/')) {
+      const ks = this.player.config.session.ks;
+      requestUrl = ks ? `${requestUrl}/ks/${ks}` : requestUrl;
+    }
+    try {
+      const response = await fetch(requestUrl);
+      const blobImage = await response.blob();
+      const href = URL.createObjectURL(blobImage);
+      return {
+        downloadUrl: href,
+        fileName: this.player.sources?.metadata?.name || 'image'
+      };
+    } catch (e: any) {}
```

Comment:
Add log here so we tell that the thumbnail could not be loaded also + the path of the reuqest - so we can debug it